server/server.go

```diff
@@ -118,6 +118,9 @@ type Server struct {
 	stats *statistics
 	// GetCertificate is used for loading certificates.
 	GetCertificate GetCertificate
+	// Sealer is called for Seal and Unseal operations.
+	// Sealer can return a gokeyless.Error to send a custom error code.
+	Sealer func(*gokeyless.Operation) ([]byte, error)
```

Comment:
I considered a named type, but there is no meaningful advantage AFAICT, and it makes the definition stutter.

As for the interface, I see the idea, but OTOH having two identical methods is weird.

If you don't have strong opinions, I'd go with what we have, otherwise I'll change it to an interface.